src/core/utils/useOnlineInternalHandleLoanReservation.ts

```diff
@@ -0,0 +1,112 @@
+import { useQueryClient } from "react-query";
+import {
+  getGetV1UserLoansQueryKey,
+  getGetV1UserReservationsQueryKey,
+  usePostV1UserLoansIdentifier,
+  usePostV1UserReservationsIdentifier
+} from "../../core/publizon/publizon";
+import { usePatronData } from "../../core/utils/helpers/usePatronData";
+import useReaderPlayer from "../../core/utils/useReaderPlayer";
+import { useUrls } from "../../core/utils/url";
+import { useModalButtonHandler } from "../../core/utils/modal";
+import { onlineInternalModalId } from "../../apps/material/helper";
+import { getAllFaustIds } from "../../core/utils/helpers/general";
+import { Manifestation } from "../../core/utils/types/entities";
+import { OnlineInternalRequestStatus } from "../../core/utils/types/request";
+import { CreateLoanResult } from "../publizon/model";
+
+type useOnlineInternalHandleLoanReservationType = {
+  manifestations: Manifestation[];
+  openModal: boolean;
+  setReservationStatus?: (status: OnlineInternalRequestStatus) => void;
+  setLoanResponse?: (response: CreateLoanResult | null) => void;
+};
+
+const useOnlineInternalHandleLoanReservation = ({
+  manifestations,
+  openModal,
+  setReservationStatus,
+  setLoanResponse
+}: useOnlineInternalHandleLoanReservationType) => {
+  const queryClient = useQueryClient();
+  const u = useUrls();
+  const authUrl = u("authUrl");
+  const { openGuarded } = useModalButtonHandler();
+
+  const { mutate: mutateLoan } = usePostV1UserLoansIdentifier();
+  const { mutate: mutateReservation } = usePostV1UserReservationsIdentifier();
+  const { data: userData } = usePatronData();
+
+  const { canBeLoaned, canBeReserved, identifier } =
+    useReaderPlayer(manifestations);
+
+  const handleModalLoanReservation = () => {
+    if (openModal) {
+      openGuarded({
+        authUrl,
+        modalId: onlineInternalModalId(getAllFaustIds(manifestations))
+      });
+      return;
+    }
+
+    if (canBeLoaned && identifier) {
+      mutateLoan(
+        { identifier },
+        {
+          onSuccess: (res) => {
+            // Ensure that the button is updated after a successful loan
+            queryClient.invalidateQueries(getGetV1UserLoansQueryKey());
+            if (setReservationStatus) {
+              setReservationStatus("loaned");
+            }
+            if (setLoanResponse) {
+              setLoanResponse(res);
+            }
+          },
+          onError: () => {
+            if (setReservationStatus) {
+              setReservationStatus("error");
+            }
+          }
+        }
+      );
+      return;
+    }
+
+    if (canBeReserved && identifier && userData?.patron) {
+      mutateReservation(
+        {
+          identifier,
+          data: {
+            email: userData.patron.emailAddress,
+            // Only add phone number if it exists
+            // Still waiting for the API to support optional phoneNumber
+            ...(userData.patron.phoneNumber && {
```

Comment:
Consider rewording this comment.

"We add the phone number if it exists" implies that the phone number is optional. "Still waiting for the API to support optional phoneNumber" implies that is it not.